### DISCUSSION TITLE
z-index to z-buffer depth question

### DISCUSSION THREAD

**Discussion by YTN0** [mikke89/RmlUi#356] [General]
How is the z-index converted to a z-buffer depth value? I am trying to interleave RmlUi components with standard 3D stuff, but I am having an issue getting the depth to work correctly.

I have an RmlUi background (z-index -100), and over that I would like to have a 3D object (at z-buffer depth 0.5f i.e. midpoint) and over my component I would like to have a window (z-index 100). However, I can only seem to get either my component on top of all RmlUi components or vice-versa.

It would be great if the z-index values mapped to specific depth levels depending on what the projection nearZ and farZ were set to, but that doesn't seem to be the case based on my testing.

**Comment by mikke89** [comment]
We don't use a z-buffer at all. If you want to render to the z-buffer then that must be done all on the client side with your renderer.

Instead of using the z-buffer with RmlUi, I would suggest that you render your 3d-scene in the correct order, between the desired RmlUi render calls. You can use a custom renderer, custom element, or a designated texture name to achieve this.
